### ISSUE TITLE
Support vector string indexing

### ISSUE THREAD

**Issue by Dheu3** [aardappel/lobster#262]
```
let s = ["hello"]
let x = int2{1,0}
print s[0][1]
print s[x]
```

Expected that line three and four would be the same, printing 101.

**Comment by aardappel** [comment]
Yes, that generally works across N vectors, but strings are not vectors, they are their own thing :)

I suppose we could enable this, but it be highly specialized, requiring its own codegen etc, so I am fine leaving it as is.
